### PR TITLE
Create .gitlab-ci.yml

### PR DESCRIPTION
**Pull request by krzysztof-wasieczko**
Added two stages for Gitlab CI/CD.
First one creates virtual machines with the following OSes: rhel9, debian11, ubuntu1804, ubuntu2004, ubuntu2204 and centos8. Second stage clones dattobd from GitHub into VMs, and builds it. Building on debian11 is temporarily disabled.

### CODE REVIEW

File: .ci/build_dattobd.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  needs: ["create:ubuntu1804:vm"]
+  script:
+    - apt-get update -y
+    - apt-get install sshpass -y
+    - sshpass -p "$VM_PASSWORD" ssh -o StrictHostKeyChecking=no datto@$UBUNTU1804_VM_IP '
+        sudo rm -rf dattobd;
+        git clone https://github.com/krzysztof-wasieczko/dattobd.git --recursive;
+        cd dattobd;
+        ls;
+        sudo yum install kernel-devel-$(uname -r) -y;
```
Comment (natalia-zelazna) [inline .ci/build_dattobd.gitlab-ci.yml:59]: yum won't work on Ubuntu :)

Comment (krzysztof-wasieczko) [reply .ci/build_dattobd.gitlab-ci.yml]: And it doesn't even fail a job in GitLab pipelines :) Removed, looks like ubuntu doesn't need any additional installations

File: .ci/build_dattobd.gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#
+# Build dattobd on RHEL9
```
Comment (Swistusmen) [inline .ci/build_dattobd.gitlab-ci.yml:2]: remove comments, build:dattobd:on:rhel9 is enough, same for the rest

File: .ci/build_dattobd.gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  stage: build dattobd
+  tags:
+    - docker-throttled
+  needs: ["create:rhel9:vm"]
+  script:
+    - apt-get update -y
+    - apt-get install sshpass -y
+    - sshpass -p "$VM_PASSWORD" ssh -o StrictHostKeyChecking=no datto@$RHEL9_VM_IP '
+        sudo rm -rf dattobd;
+        git clone https://github.com/krzysztof-wasieczko/dattobd.git --recursive;
```
Comment (Swistusmen) [inline .ci/build_dattobd.gitlab-ci.yml:14]: why it's ur private repo, not the official one?

File: .ci/build_dattobd.gitlab-ci.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        cd dattobd;
+        ls;
```
Comment (Swistusmen) [inline .ci/build_dattobd.gitlab-ci.yml:16]: remove ls

File: .ci/build_dattobd.gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  needs: ["create:debian11:vm"]
+  script:
+    - apt-get update -y
+    - apt-get install sshpass -y
+    - sshpass -p "$VM_PASSWORD" ssh -o StrictHostKeyChecking=no datto@$DEBIAN11_VM_IP '
+        sudo rm -rf dattobd;
+        git clone https://github.com/krzysztof-wasieczko/dattobd.git --recursive;
+        cd dattobd;
+        ls'
+#        sudo yum install kernel-devel-$(uname -r) -y;
```
Comment (Swistusmen) [inline .ci/build_dattobd.gitlab-ci.yml:39]: on debian it's apt not yum

File: .ci/create_vm.gitlab-ci.yml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+# Create VM with RHEL9
+#
+create:rhel9:vm:
+  stage: create vm
+  tags:
+    - docker-throttled
+  script:
+    - echo "TEST HAS STARTED"
+    - apt-get update -y
+    - apt-get install curl -y
```
Comment (Swistusmen) [inline .ci/create_vm.gitlab-ci.yml:11]: on rhel 9 it's yum, not apt

File: .ci/create_vm.gitlab-ci.yml [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    - echo "Creating virtual machine with RHEL9"
+    - 'curl --location "http://qapi.datto.net/v1/agent" --header "Content-Type: application/json" --data "{\"os\":\"rhel9\",\"agent_type\":\"\",\"region\":\"nwk\",\"ttl\":1}" > createVmOutput.txt'
+    - cat createVmOutput.txt
+    - OUTPUT=$(cat createVmOutput.txt)
+    - echo $OUTPUT
```
Comment (Swistusmen) [inline .ci/create_vm.gitlab-ci.yml:16]: printing 2 times creatVmOutput is not needed

File: .ci/create_vm.gitlab-ci.yml [hunk 8]
```diff
@@ -0,0 +1,161 @@
+#
+# Create VM with RHEL9
```
Comment (Swistusmen) [inline .ci/create_vm.gitlab-ci.yml:2]: comment is not needed as create:rhel9:vm provides info of what is happening

File: .ci/create_vm.gitlab-ci.yml [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+#
+# Create VM with CENTOS8
+#
+create:centos8:vm:
+  stage: create vm
+  tags:
+    - docker-throttled
+  script:
+    - echo "TEST HAS STARTED"
+    - apt-get update -y
```
Comment (Swistusmen) [inline .ci/create_vm.gitlab-ci.yml:145]: yum?

File: .ci/build_dattobd.gitlab-ci.yml (hunk 5 above)
Comment (krzysztof-wasieczko) [reply .ci/build_dattobd.gitlab-ci.yml]: I removed it totally cause we know that anyway on debian it won't build right know, and maybe installing those it's not even needed.

File: .ci/build_dattobd.gitlab-ci.yml (hunk 4 above)
Comment (krzysztof-wasieczko) [reply .ci/build_dattobd.gitlab-ci.yml]: Done

File: .ci/build_dattobd.gitlab-ci.yml (hunk 3 above)
Comment (krzysztof-wasieczko) [reply .ci/build_dattobd.gitlab-ci.yml]: Good question , probably it's leftover from my tests. I've changed to original one.

File: .ci/create_vm.gitlab-ci.yml (hunk 6 above)
Comment (krzysztof-wasieczko) [reply .ci/create_vm.gitlab-ci.yml:11]: In this step we are not on RHEL9 yet, VM with RHEL is not created yet. We are on docker executor which is ubuntu16.04 and we're installing curl on this ubuntu image to send request to create RHEL9 VM.

File: .ci/create_vm.gitlab-ci.yml (hunk 7 above)
Comment (krzysztof-wasieczko) [reply .ci/create_vm.gitlab-ci.yml:12]: done

File: .ci/create_vm.gitlab-ci.yml [hunk 10]
```diff
@@ -0,0 +1,137 @@
+create:rhel9:vm:
+  stage: create vm
+  tags:
+    - docker-throttled
+  script:
+    - echo "TEST HAS STARTED"
```
Comment (Swistusmen) [inline .ci/create_vm.gitlab-ci.yml:6]: remove this

Comment (krzysztof-wasieczko) [reply .ci/create_vm.gitlab-ci.yml]: done

**Review by Swistusmen — APPROVED** [review]

**Review by natalia-zelazna — APPROVED** [review]

**Pull request merged**